### ISSUE TITLE
Have HashMap and BTreeMap implement FromPyObject

### ISSUE THREAD

**Issue by bentheiii** [PyO3/pyo3#516]
It would be relatively easy to allow for HashMap and BTreeMap arguments by having those types implement FromPyObject. A sample implementation might look like this:

```rust
impl<'source, K, V> FromPyObject<'source> for HashMap<K, V>
    where K: FromPyObject<'source>, V: FromPyObject<'source> {
    fn extract(ob: &'source PyAny) -> Result<Self, PyErr> {
        let dict = <PyDict as PyTryFrom>::try_from(ob)?;
        let mut ret = HashMap::new();
        for (k, v) in dict.iter() {
            ret.insert(K::extract(k)?, V::extract(v)?);
        }
        Ok(ret)
    }
}
```

**Comment by gdesmott** [comment]
I created a PR based on your suggested code: https://github.com/PyO3/pyo3/pull/702

**Comment by Alexander-N** [comment]
Closed by #702
